content/00.build/65.developer-reference/40.account-abstraction/50.signature-validation.md

```diff
@@ -106,3 +158,22 @@ bytes memory input = abi.encodePacked(
 
 You can find a more in-depth example showing how
 it can be used in the ["Signing Transactions with WebAuthn"](https://code.zksync.io/tutorials/signing-transactions-with-webauthn) tutorial.
+
+## Offchain Signature Verification
+
+The `zksync-ethers` SDK provides two methods within `utils` to verify standard signatures of an account:
+[`isMessageSignatureCorrect`](https://sdk.zksync.io/js/ethers/api/v6/utilities#ismessagesignaturecorrect) and [`isTypedDataSignatureCorrect`](https://sdk.zksync.io/js/ethers/api/v6/utilities#istypeddatasignaturecorrect).
+
+```ts
+import { utils, EIP712Signer } from "zksync-ethers";
+
+const isValidMessageSignature = await utils.isMessageSignatureCorrect(provider, ADDRESS, message, messageSignature);
+
+const isValidTypesSignature = await utils.isTypedDataSignatureCorrect(provider, ADDRESS, await eip712Signer.getDomain(), utils.EIP712_TYPES, EIP712Signer.getSignInput(tx), typedSignature);
+```
+
+Currently these methods only support verifying ECDSA signatures, but very soon they will support EIP1271 signature verification as well.
+
+Both of these methods return `true` or `false` depending on whether the message signature is correct.
+
+It is **not recommended** to use the `ethers.js` library to verify user signatures.
```

Comment:
why?

Reply:
good question, I updated this to explain why